### ISSUE TITLE
not working well with Kubernetes 1.19

### ISSUE THREAD

**Issue by wansir** [kubesphere/kubesphere#2928]
**Describe the Bug**
[k8s update golang to 1.15](https://github.com/kubernetes/kubernetes/pull/93264), self-signed certificate for admission webhook [needs to be updated](https://github.com/golang/go/issues/39568).

```
x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0, requeuing
```

**Versions Used**
KubeSphere: v3.0.0
Kubernetes: v1.19.0

/kind bug

**Comment by wenerme** [comment]
this is not related to k8s v1.19, but go v1.15

```bash
# workaround
export GODEBUG=x509ignoreCN=0
```

but kubesphere is not compact with v1.19 csr v1, devops deploy will never success.

```json
{
  "changed": true,
  "cmd": "/bin/bash /kubesphere/kubesphere/s2i/certs.sh --service webhook-server-service --namespace kubesphere-devops-system --secret webhook-server-secret \n",
  "delta": "0:00:04.242800",
  "end": "2020-10-28 00:22:02.518310",
  "msg": "non-zero return code",
  "rc": 1,
  "start": "2020-10-28 00:21:58.275510",
  "stderr": "Generating RSA private key, 2048 bit long modulus (2 primes)\n........................................................................................................................+++++\n..............................................................................................................................................................+++++\ne is 65537 (0x010001)\nerror: no kind \"CertificateSigningRequest\" is registered for version \"certificates.k8s.io/v1\" in scheme \"k8s.io/kubectl/pkg/scheme/scheme.go:28\"",
  "stderr_lines": [
    "Generating RSA private key, 2048 bit long modulus (2 primes)",
    "........................................................................................................................+++++",
    "..............................................................................................................................................................+++++",
    "e is 65537 (0x010001)",
    "error: no kind \"CertificateSigningRequest\" is registered for version \"certificates.k8s.io/v1\" in scheme \"k8s.io/kubectl/pkg/scheme/scheme.go:28\""
  ],
  "stdout": "creating certs in tmpdir /tmp/tmp.adhIaE \ncertificatesigningrequest.certificates.k8s.io \"webhook-server-service.kubesphere-devops-system\" deleted\ncertificatesigningrequest.certificates.k8s.io/webhook-server-service.kubesphere-devops-system created\nNAME                                              AGE   SIGNERNAME                     REQUESTOR                                              CONDITION\nwebhook-server-service.kubesphere-devops-system   1s    kubernetes.io/legacy-unknown   system:serviceaccount:kubesphere-system:ks-installer   Pending\nNo resources found",
  "stdout_lines": [
    "creating certs in tmpdir /tmp/tmp.adhIaE ",
    "certificatesigningrequest.certificates.k8s.io \"webhook-server-service.kubesphere-devops-system\" deleted",
    "certificatesigningrequest.certificates.k8s.io/webhook-server-service.kubesphere-devops-system created",
    "NAME                                              AGE   SIGNERNAME                     REQUESTOR                                              CONDITION",
    "webhook-server-service.kubesphere-devops-system   1s    kubernetes.io/legacy-unknown   system:serviceaccount:kubesphere-system:ks-installer   Pending",
    "No resources found"
  ]
}
```

**Comment by wansir** [comment]
Use the following command to solve this problem:

```
kubectl apply -f https://raw.githubusercontent.com/kubesphere/ks-installer/2c4b479ec65110f7910f913734b3d069409d72a8/roles/ks-core/prepare/files/ks-init/users.iam.kubesphere.io.yaml
kubectl apply -f https://raw.githubusercontent.com/kubesphere/ks-installer/2c4b479ec65110f7910f913734b3d069409d72a8/roles/ks-core/prepare/files/ks-init/webhook-secret.yaml
kubectl -n kubesphere-system rollout restart deploy ks-controller-manager
```

**Comment by kele1997** [comment]
Versions Used
KubeSphere: v3.0.0
Kubernetes: v1.19.0

the same problems with **logsidecar-injector**, I try to modify `logsidecar-injector-admission-certs` secret, replace `sever.crt` with `tls.crt`, `sever.key` with `tls.key`, but doesn't work.  any idea about this? @wansir 



```
Internal error occurred: failed calling webhook "logsidecar-injector.logging.kubesphere.io": Post "https://logsidecar-injector-admission.kubesphere-logging-system.svc:443/?timeout=30s": x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0
```

**Comment by caihy05** [comment]
@wansir 解决问题，感谢，估计是安装过程中存储问题导致

**Comment by zryfish** [comment]
fixed on v3.1
/close
